invenio_app_ils/circulation/search.py

```diff
@@ -138,3 +113,36 @@ def get_most_loaned_documents(from_date, to_date, bucket_size):
     search = search[:0]
 
     return search
+
+
+def get_loan_states_aggregation(document_pid, states, patron_pid=None):
```

Comment:
Since the method is returning a search query/obj, I would rename the method and change the explanation to something similar to:
```suggestion
def get_loans_aggregated_by_states(document_pid, states, patron_pid=None):
```

```
     """Returns loans aggregated by states for a given document.""" 
```